lib/marklogic.js

```diff
@@ -716,6 +716,20 @@ MarkLogicClient.prototype.createProxy = function createProxy(serviceDeclaration)
   return proxy.init(this, serviceDeclaration);
 };
 
+MarkLogicClient.prototype.serviceCaller = function serviceCaller(serviceDeclaration, endpointDeclarations) {
+  if(serviceDeclaration == null || endpointDeclarations == null) {
+    throw new Error('Parameters - serviceDeclaration and endpointDeclarations are required');
+  }
```

Comment:
Given that the ServiceCaller constructor validates, it's might be unnecessary to validate here.